chemprop/featurizers/molecule.py

```diff
@@ -41,3 +49,68 @@ class MorganBinaryFeaturizer(MorganFeaturizerMixin, BinaryFeaturizerMixin, Vecto
 @MoleculeFeaturizerRegistry("morgan_count")
 class MorganCountFeaturizer(MorganFeaturizerMixin, CountFeaturizerMixin, VectorFeaturizer[Mol]):
     pass
+
+
+@MoleculeFeaturizerRegistry("rdkit_2d")
+class RDKit2DFeaturizer(VectorFeaturizer[Mol]):
+    def __init__(self):
+        warnings.warn(
+            "The RDKit 2D features can deviate signifcantly from a normal distribution. Consider "
+            "manually scaling them using an appropriate scaler before creating datapoints, rather "
+            "than using the scikit-learn `StandardScaler` (the default in Chemprop)."
+        )
+
+    def __len__(self) -> int:
+        return len(Descriptors.descList)
+
+    def __call__(self, mol: Chem.Mol) -> np.ndarray:
+        features = np.array(
+            [
+                func(mol)
+                for _, func in filter(
+                    lambda i: i[0] != "SPS" or mol.GetNumHeavyAtoms() > 0, Descriptors.descList
+                )
+            ],
+            dtype=float,
+        )
+
+        return features
+
+
+if not NO_DESCRIPTASTORUS:
+
+    class V1RDKit2DFeaturizerMixin(VectorFeaturizer[Mol]):
+        def __len__(self) -> int:
+            return 200
+
+        def __call__(self, mol: Mol) -> np.ndarray:
+            smiles = Chem.MolToSmiles(mol, isomericSmiles=True)
+            features = self.generator.process(smiles)[1:]
+
+            return np.array(features)
+
+    @MoleculeFeaturizerRegistry("v1_rdkit_2d")
+    class V1RDKit2DFeaturizer(V1RDKit2DFeaturizerMixin):
+        def __init__(self):
+            self.generator = rdDescriptors.RDKit2D()
+
+    @MoleculeFeaturizerRegistry("v1_rdkit_2d_normalized")
+    class V1RDKit2DNormalizedFeaturizer(V1RDKit2DFeaturizerMixin):
+        def __init__(self):
+            self.generator = rdNormalizedDescriptors.RDKit2DNormalized()
+
+else:
+
+    @MoleculeFeaturizerRegistry("v1_rdkit_2d")
+    class V1RDKit2DFeaturizer:
+        """Mock implementation raising an ImportError if descriptastorus cannot be imported."""
+
+        def __init__(self):
+            raise ImportError(
+                "Failed to import descriptastorus. Please install descriptastorus "
+                "(https://github.com/bp-kelley/descriptastorus) to use RDKit 2D features."
+            )
```

Comment:
I saw that this is a mock implementation. You might already have an idea of how to implement it in a real implementation. Here is my two cents:
Move this into `V1RDKit2DFeaturizerMixin` and check if `NO_DESCRIPTASTORUS` is True.

Reply:
Good idea